### PR TITLE
[NETBEANS-438] Cannot Sign Platform app due to symlink

### PR DESCRIPTION
**Pull request by SirIntellegence**
Now creating an intermediate sh file to invoke the executable so resulting apps will sign properly

### CODE REVIEW

**Comment by SirIntellegence** [comment]
Could someone review this? Maybe @matthiasblaesing or @JaroslavTulach?

File: apisupport.harness/release/suite.xml [hunk 1]
```diff
@@ -420,7 +420,17 @@ See: http://wiki.netbeans.org/DevFaqHowToReuseModules
               tofile="${dist.dir}/${app.name}.app/Contents/Resources/${app.name}.icns"/>
         
         <delete file="${dist.dir}/${app.name}.app/Contents/MacOS/${app.name}"/>
-        <symlink link="${dist.dir}/${app.name}.app/Contents/MacOS/${app.name}" resource="../Resources/${app.name}/bin/${app.name}"/>
+        <echo file="${dist.dir}/${app.name}.app/Contents/MacOS/${app.name}">#!/bin/sh
```
Comment (JaroslavTulach) [inline apisupport.harness/release/suite.xml]: I am a Linux user, maybe our Mac users @tzezula or @thurka can help with the review.

Comment (SirIntellegence) [reply apisupport.harness/release/suite.xml]: Have they been notified of the request by you referencing them? I haven't noticed anyone checking this change yet. It does work for me, if that means anything.

**Comment by emilianbold** [comment]
I'll check this next week.

**Comment by lkishalmi** [comment]
This one is open for a while. Can somebody confirm that this works?

**Comment by dtrebbien** [comment]
Can you please post steps that would enable a macOS user to try out this patch? Walking through creating a "Hello, world!"-type app would probably suffice, leaving out the steps to actually sign the app.

What is the purpose of this:

```
cd "${0%/*}"
```

Also, I am confused by the comment "#make absolute and resolve symlinks" because the next line is a `cd` command.

Can the three `cd`s be combined?

Would something like this work:

```xml
<echo file="${dist.dir}/${app.name}.app/Contents/MacOS/${app.name}">#!/bin/sh
../Resources/"${app.name}"/bin/"${app.name}" "$$@"
</echo>
```

(I am not sure if this would affect dynamically linking DYLIBs that are included in the app.)

**Comment by SirIntellegence** [comment]
@dbalek The first `cd` is to `cd` to the location of the script under `Contents/MasOS` since we are not likely to be in that directory. The second is to cd to the dir with the actual script. The third one cd's to the result of `pwd -P`, which makes the current path absolute and resolves all symlinks. That last one is done in the main script, but I felt it might be better to do that using `pwd` than to use the logic in the main executable since that is mostly what `pwd` is for.

Also, to test this, you will need to make a Netbeans Platform app, build it with "build-mac", and then sign the result with `codesign`. See https://developer.apple.com/library/content/documentation/Security/Conceptual/CodeSigningGuide/Procedures/Procedures.html to learn how to use `codesign`

**Comment by neilcsmith-net** [comment]
While I understand what the `pwd -P` is doing, isn't the way @dtrebbien suggests simpler and possibly preferred - at the moment, unless I'm missing something, the launch script doesn't `cd` before executing?

**Comment by SirIntellegence** [comment]
The main issue with the suggested approach is that if we execute the script in `~`, it will look for the executable at `~/../Resources/"${app.name}"/bin/"${app.name}"`, so you need to `cd` to the directory first. The symlink works, but `codesign` doesn't like it when the "main executable" is a symlink, so the one I provided `cd`s to `../Resources/"${app.name}"/bin`.
As for what `${0%/*}` is, see https://stackoverflow.com/q/6393551/1366594
I did just notice that I try to change to the bin dir before getting the real directory of the script. That might need to be tested in the case that the new mac bin script is symlinked.

**Comment by neilcsmith-net** [comment]
@SirIntellegence oops, yes, missed that.  But maybe it needs to resolve to an absolute path without `cd` ing to the directory?  I'm not sure of the implication of changing directory, but that would be a behavioural change I think - the launcher doesn't `cd` from what I remember, it resolves to an absolute path to execute.  Can't we do same here to resolve, copying the existing path resolution from the top of the launcher?

**Comment by SirIntellegence** [comment]
There we go, will that work?

**Comment by SirIntellegence** [comment]
@emilianbold Could you check the new changes? I have updated it so it doesn't cd.

**Comment by emilianbold** [comment]
Cloned your branch, let me see how to sign...

**Comment by emilianbold** [comment]
I confirm that I'm able to codesign the resulting app with these patches.

Without the patches `codesign` complains that: `the main executable or Info.plist must be a regular file (no symlinks, etc.)`.

This took me forever to verify because my Apple certificate had expired and then the SirIntellegence:NETBEANS-438 branch did not have 
commit 74bb76a60a53439551a76f1e7d123925d5d5fa55 (`More robustness when resource name of loadImageIcon is null`) and I was getting weird exceptions that made me believe that perhaps my hard drive is failing?

The shell code looks a bit complicated, some comments would be nice.

Other than that +1

**Pull request merged**